src/main/groovy/org/transmartproject/core/biomarker/BioMarkerConstraint.groovy

```diff
@@ -0,0 +1,34 @@
+package org.transmartproject.core.biomarker
+
+/**
+ * Limits the results of {@link BioMarkerResource#retrieveBioMarkers(List<BioMarkerConstraint>)}
+ *
+ * Properties constraints are passed as map where key represents property name and value represents the value
+ * that property must be equal to. When map contains multiple properties constraints all of them must be met by
+ * bio marker entry to appear in the result. In other words it's assumed there is AND logical operation
+ * between property constraints.
+ * e.g. [type: 'PROTEIN', organism: 'HOMO SAPIENS'] limits result to the human proteins
+ *
+ * If collection is passed as property value the given property of result bio marker has to match
+ * to one of the supplied values
+ * e.g. [type: 'GENE', name: ['TP53', 'AURKA', 'ADIRF']]
+ * limits results to three specified genes.
+ */
+interface BioMarkerConstraint {
+
+    /**
+     * Constraint that limits the bio markers in the result to those that have
+     * a specific values for a certain properties.
+     *
+     * Property names (see {@link BioMarker}): id, type, primaryExternalId, primarySourceCode, name, description, organism
+     */
+    public final static String PROPERTIES_CONSTRAINT = 'propertiesConstraint';
+
+    /**
+     * Constraint that limits the bio markers in the result to those that have
+     * associated bio marker with a specific values for a certain properties.
+     *
+     * Property names (see {@link BioMarker}): id, type, primaryExternalId, primarySourceCode, name, description, organism
+     */
+    public final static String ASSOC_BIO_MARKER_PROPERTIES_CONSTRAINT = 'assocBioMarkerPropertiesConstraint';
```

Comment:
A constraint that took as parameter the subconstraints for the biomarkers for which associated biomarkers were wanted would maximize composability. E.g. `createConstraint(ASSOC_BIO_MARKER_PROPERTIES_CONSTRAINT, subconstraints: [ (PROPERTIES_CONSTRAINT): [ type: 'GENE' ]])` would get all the biomarkers associated with gene biomarkers. This would probably be much more difficult to implement though, so this may be fine for now.